### DISCUSSION TITLE
Does wallet connect v2 support hardhat chain id 1337?

### DISCUSSION THREAD

**Discussion by johnnyleejy** [WalletConnect/walletconnect-monorepo#2868] [Developer Support] [answered]
When creating my project on wallet connect, I saw a drop down list of supported chain ids and 1337 is not one of them. 

<img width="886" alt="image" src="https://github.com/WalletConnect/walletconnect-monorepo/assets/108564365/704abe9a-51b0-4a44-9f03-216838b9bd4a">

In my code, I am specifying
```
    const provider = await EthereumProvider.init({
      projectId: '<project_id>',
      chains: [1337],
      showQrModal: true,
      rpcMap: {
        1337: getProvidersUrl(),
      },
      qrModalOptions: {
        enableExplorer: false,
      },
    });

    await provider.enable();
```

And the above code is not triggering a propose session event to my localhost wallet. 

Wondering if its an issue regarding the chain id?

Thank you.

**Comment by Cali93** [comment]
WalletConnect supports EVM chains by default. 
It's very likely that it's a misconfiguration issue with `EthereumProvider.init` params.
The explorer form in your screenshot is purely informative and will allow other people to discover which chains your project supports so it's not related.
Could you provide more context about the errors you're seeing in the console ?

**Reply by johnnyleejy** [reply]
Thanks for your reply @Cali93 , there are no errors. It's just that my web3Wallet is not picking up any `session_proposal` events.

Previously, using `WalletConnectProvider` v1, all I need to to call `provider.enable()` and my wallet can pick up the session request event.

**Reply by Cali93** [reply]
Can you share the package versions you're using for `Web3Wallet`, `EthereumProvider` and any other @walletconnect related libraries please ?

**Reply by johnnyleejy** [reply]
For my mock dapp, i am using:
1. "@walletconnect/ethereum-provider": "^2.8.5"
2. "@walletconnect/modal": "^2.5.6"


For my wallet, I am using:
1. "@walletconnect/web3wallet": "^1.5.1"


Other libraries including wallet connect v1.

1. "@walletconnect/client": "^1.8.0",
2. "@walletconnect/core": "^2.6.1",
3. "@walletconnect/legacy-types": "^2.0.0",
4. "@walletconnect/utils": "^2.6.1",
5. "@walletconnect/web3-provider": "^1.8.0",

**Reply by johnnyleejy** [reply]
To add more details, here's the flow.

1. Go to Dapp and open Wallet Connect QR Modal.
2. Copy QR code link from Dapp
3. Paste link in web3Wallet and connect
4. web3Wallet attempts to pair with uri by using
`await web3wallet.pair({ uri });`
5. Nothing happens. the below details are logged out.

<img width="540" alt="image" src="https://github.com/WalletConnect/walletconnect-monorepo/assets/108564365/0284f102-4e8b-491f-bff0-443dfcea97f3">

**Reply by Cali93** [reply]
I think you should update every dependency to latest (eg: `npm i @walletconnect/ethereum-provider@latest @walletconnect/modal@latest` etc.) for both the dapp and the wallet and make sure to follow the snippets provided in the documentation regarding configuration.
You should also uninstall dependencies related to WalletConnect V1 (refer to the documentation for installing the dependencies that you need).
Here's an [example dapp using v2](https://github.com/WalletConnect/web3modal-examples/tree/main/walletconnect-modal-ethereum-provider-react).
For the wallet you can refer to [Web3Wallet docs](https://docs.walletconnect.com/2.0/web/web3wallet/installation).
If the issue still persists after that, let us know and we'll gladly help you.

**Reply by johnnyleejy** [reply]
Thanks @Cali93 , I am still having issues connecting to my wallet but managed to narrow down the issue.

My session proposal from my dapp is stuck in pending and my web3Wallet never handles the proposal event.

I have registered an `session_proposal` event listener for my web3wallet but it never picked up any session proposal event from my dapp.

My wallet is able to pick up session proposal events from other dapps such as openSea, uniswap etc.

**Side note**: **BOTH** my dapp and and wallet are hosted on localhost:3000 and i followed exactly the code given [here](https://github.com/WalletConnect/web3modal-examples/blob/main/walletconnect-modal-ethereum-provider-react/src/pages/index.jsx).

Logging out the results using `web3wallet.getPendingSessionProposals()` is the stuck session proposal from my dapp:

```
[
  {
    "id": 1688223933008921,
    "expiry": 1688224233,
    "requiredNamespaces": {
      "eip155": {
        "chains": [
          "eip155:1"
        ],
        "methods": [
          "eth_sendTransaction",
          "personal_sign"
        ],
        "events": [
          "chainChanged",
          "accountsChanged"
        ],
        "rpcMap": {
          "1": "https://rpc.walletconnect.com/v1/?chainId=eip155:1&projectId=89080488436c9df43eebf70e7e490918"
        }
      }
    },
    "optionalNamespaces": {
      "eip155": {
        "chains": [
          "eip155:1337"
        ],
        "methods": [
          "eth_sendTransaction",
          "personal_sign"
        ],
        "events": [
          "chainChanged",
          "accountsChanged"
        ],
        "rpcMap": {
          "1": "https://rpc.walletconnect.com/v1/?chainId=eip155:1&projectId=89080488436c9df43eebf70e7e490918",
          "1337": "https://rpc.walletconnect.com/v1/?chainId=eip155:1337&projectId=89080488436c9df43eebf70e7e490918"
        }
      }
    },
    "relays": [
      {
        "protocol": "irn"
      }
    ],
    "proposer": {
      "publicKey": "c511e204344afa102376dc56ea9e8d68b0283590759e38dba999410b4c8ab74c",
      "metadata": {
        "description": "",
        "url": "http://localhost:3000",
        "icons": [
          "http://localhost:3000/favicon.ico"
        ],
        "name": "Test dapp"
      }
    }
  }
]
```

**Reply by johnnyleejy** [reply]
I tested my mock dapp with the [react-example](https://react-web3wallet.vercel.app/walletconnect) and it's working.

Problem is with my web3Wallet here.

I am using the below code and my web3Wallet can log out session proposals from most dapps EXCEPT my mock dapp.

But my mock dapp is able to connect with the react example.

**Edit**: Not sure if v2 allows dapp and wallet pairing on the same host port. My pairing worked when using v1 but I suspect there are pairing issues when proposing and pairing on the same host and port.

**UPDATE**: I cloned https://react-web3wallet.vercel.app/walletconnect and added a button to use `@walletconnect/ethereum-provider` to init and connect to the web3Wallet example. And it doesn't work. It seems like you cannot propose a session and listen to the session on the same host and port. In my case, it is localhost:3000.

**UPDATE 2:** I tried messing with the localstorage storage, clearing them and refreshing my page and it works. Not sure if this is a walletconnect bug or what?

```
    web3wallet.on("session_proposal", async (proposal) => {
        const session = await web3wallet.approveSession({
            id: proposal.id,
            namespaces,
        });
    });
    await web3wallet.core.pairing.pair({ uri })
```

**Reply by johnnyleejy** [reply]
After even more testing, here's my hypothesis.

The dapp and wallet connection on localhost:3000 only works if the local storage only has the keychain key value pair.

<img width="798" alt="image" src="https://github.com/WalletConnect/walletconnect-monorepo/assets/108564365/79ef10cc-2020-40ee-9065-4f6864af6076">

It will not work in any other states like the one below.
<img width="798" alt="image" src="https://github.com/WalletConnect/walletconnect-monorepo/assets/108564365/cf591b3c-628d-43de-9a0f-a396bb31eb56">


This issue seems to be related to the local storage and how the `ethereumProvider` and `web3Wallet` is reading, writing these values.

**Reply by ganchoradkov** [reply]
Hey, what does that mean exactly `The dapp and wallet connection on localhost:3000 `? 
Both dapp & wallet working on the same web page or the same app?

**Reply by johnnyleejy** [reply]
@ganchoradkov , same web page. 

You may refer to https://github.com/WalletConnect/walletconnect-monorepo/issues/2975 for the video of the issue i recorded.

**Reply by ganchoradkov** [reply]
@johnnyleejy This functionality unfortunately is not yet supported. The clients are registered per connection and in your case, you're attempting to use a single connection for both sending & receiving requests which is bugging out. 
Is it possible to separate the instances of the dapp & wallet to different ports?

**Comment by johnnyleejy** [comment, marked as answer]
Thanks @ganchoradkov , separating the instances is our last resort and it seems that we will have to do it this way. Hope this functionality gets implemented soon!
